scripts/report.sh

```diff
@@ -0,0 +1,184 @@
+#!/bin/bash
+
+oc_installed=false
+kubectl_installed=false
+platform="kubectl"
+
+oc >/dev/null
+
+if [ $? -eq 0 ]; then
+	oc_installed=true
+	platform="oc"
+fi
+
+kubectl >/dev/null
+
+if [ $? -eq 0 ]; then # we will use kubectl with priority (?)
+	kubectl_installed=true
+	platform="kubectl"
+fi
+
+if [[ $oc_installed = false && $kubectl_installed = false ]]; then
+	echo "There is no oc or kubectl installed"
+	exit 1
+fi
+
+# default values
+cluster="my-cluster"
+namespace="myproject"
+
+usage() {
+	echo "Usage: $0 [--namespace <string>] [--cluster <string>]" 1>&2; 
+	echo "Default values namespace=my-project cluster=my-cluster" 1>&2; 
+	exit 1; 
+}
+
+optspec=":-:"
+while getopts "$optspec" optchar; do
+    case "${optchar}" in
+        -)
+            case "${OPTARG}" in
+                cluster=*)
+                    cluster=${OPTARG#*=}
+                    ;;
+                namespace=*)
+                    namespace=${OPTARG#*=}
+                    ;;
+                *)
+                    usage
+                    ;;
+            esac;;
+        *)
+            cluster="my-cluster"
+            namespace="myproject"
+            ;;
+    esac
+done
+shift $((OPTIND-1))
+
+direct=`mktemp -d`
+resources_to_fetch=(
+	"deployments"
+	"statefulsets"
+	"configmaps"
+	"secrets"
+	"services"
+	"poddisruptionbudgets"
+	"roles"
+	"rolebindings"
+	"networkpolicies"
+	"routes"
+	"pods"
+	)
```

Comment:
I was wondering if the replicasets under the ststefulsets could be useful as well, for getting more information.
What about the pvc as well?

Reply:
@stanlyDoge I think especially the PVCs are a very good point. Replica Sets should be easy to add as well.